common/src/main/java/org/opensearch/ml/common/utils/IndexUtils.java

```diff
@@ -32,4 +40,40 @@ public class IndexUtils {
     // Note: This does not include static settings like number of shards, which can't be changed after index creation.
     public static final Map<String, Object> UPDATED_DEFAULT_INDEX_SETTINGS = Map.of("index.auto_expand_replicas", "0-1");
     public static final Map<String, Object> UPDATED_ALL_NODES_REPLICA_INDEX_SETTINGS = Map.of("index.auto_expand_replicas", "0-all");
+
+    public static String getMappingFromFile(String path) throws IOException {
+        URL url = IndexUtils.class.getClassLoader().getResource(path);
+        if (url == null) {
+            throw new IOException("Resource not found: " + path);
+        }
+
+        String mapping = Resources.toString(url, Charsets.UTF_8);
+        if (mapping == null || mapping.isBlank()) {
+            throw new IllegalArgumentException("Empty mapping found at: " + path);
+        }
+
+        if (!StringUtils.isJson(mapping)) {
+            throw new JsonSyntaxException("Mapping is not a valid JSON: " + path);
+        }
```

Comment:
This line 50 Resources.toString(url, Charsets.UTF_8) never returns null. Instead, it will throw an exception if the resource cannot be read. 
With that said, I think these two if() blocks can be simplified to only 1:

String mapping = Resources.toString(url, Charsets.UTF_8).trim();
    if (mapping.isEmpty() || !StringUtils.isJson(mapping)) {
        throw new IllegalArgumentException("Invalid or non-JSON mapping at: " + path);
    }